src/construction.cpp

```diff
@@ -1377,13 +1377,13 @@ void check_constructions()
             debugmsg("Unknown skill %s in %s", c->skill.c_str(), display_name.c_str());
         }
         c->requirements.check_consistency(display_name);
-        if (!c->pre_terrain.empty() && !c->pre_is_furniture && termap.count(c->pre_terrain) == 0) {
+        if (!c->pre_terrain.empty() && !c->pre_is_furniture && !ter_str_id( c->pre_terrain ).is_valid() ) {
```

Comment:
Not a problem with this PR itself, but for future refactoring:

Usually, there should be no need to check for validity of a string id (`is_valid`). This should be done once, at the start of the game, when the game data has been loaded. Invalid ids should be reported. After that, all ids should be assumed to be valid. This makes the code simpler (no need to check for validity) and reports error early and reliable.

The null id can be used when no real id applies.

The logic here is bad (but that is a obviously a pre-existing problem). What if somebody adds a terrain "foo" *and* a furniture "foo"? The code here works because it assumes that all furniture ids and all terrain ids have a different prefix. Ideally, the JSON data would indicate exactly whether this is supposed to be a terrain or a furniture.